### DISCUSSION TITLE
Validate a field when it changes

### DISCUSSION THREAD

**Discussion by ar-dehghan-a** [mantinedev/mantine#6301] [Q&A] [answered]
I want some fields to execute the validation function as soon as their content changes and errors appear. I did this with the following code.

``` js
const handlePersianInput = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
  const path = e.target.getAttribute('data-path')!
  const value = e.target.value
  form.setFieldValue(path, value)
  form.validateField(path)
}
 ```

But the problem is that when the user writes a character in the input that is wrong for the first time, an error is displayed, but with the second character, the error is deleted. If it is still wrong, the error is displayed again with the next character and deleted again with the next character. And I continue like this. I want the information not to be deleted in the middle of an error when the user enters it wrongly

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/form/validation/#validate-fields-on-change

**Reply by ar-dehghan-a** [reply]
Thank you, I had not paid attention
